onadata/apps/api/viewsets/openid_connect_viewset.py

```diff
@@ -159,7 +161,12 @@ def create_or_get_user(
                                                    'username': username
                                                })
     if created:
-        UserProfile.objects.create(user=user)
+        property(
+            lambda u: cache_single_object(
+                USER_PROFILE_PREFIX,
+                user.id,
+                UserProfile.objects.get_or_create(user=user)
+                ))
```

Comment:
Why did we have to use a lamba function & property here ?

Reply:
An attempt to make it a getter function. I can remove them

Reply:
Mhh okay... I'm a bit confused on why we need the getter function ?